core/modules/qmeta/QStatusMysql.cc

```diff
@@ -0,0 +1,189 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "qmeta/QStatusMysql.h"
+
+// System headers
+#include <algorithm>
+
+// Third-party headers
+#include "boost/lexical_cast.hpp"
+#include <boost/algorithm/string/replace.hpp>
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qmeta/Exceptions.h"
+#include "qmeta/QMetaTransaction.h"
+#include "sql/SqlResults.h"
+
+
+namespace {
+
+// Current version of QStatus schema.
+char const VERSION_STR[] = "0";
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.qmeta.QStatusMysql");
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace qmeta {
+
+
+QStatusMysql::QStatusMysql(mysql::MySqlConfig const& mysqlConf)
+  : QStatus(), _conn(mysqlConf) {
+    // Check that database is in consistent state
+    try {
+        createQueryStatsTmpTable();
+    } catch (SqlError const& e) {
+        LOGS(_log, LOG_LVL_WARN, "QStatusMysql create table QStatsTmp failed. " << e.what());
+    }
+}
+
+
+void QStatusMysql::createQueryStatsTmpTable() {
+    LOGS(_log, LOG_LVL_DEBUG, "QStatusMysql create, if not exists, table QStatsTmp");
+    // If this is being run outside the constructor, _bdMutex may need to be locked first.
+    // Try to create the temporary table if it is not there.
+    QMetaTransaction trans(_conn);
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    std::string query = "CREATE TABLE IF NOT EXISTS QStatsTmp (queryId bigint(20), "
+                        "totalChunks int, completedChunks int, "
+                        "queryBegin timestamp DEFAULT 0, "
+                        "lastUpdate timestamp DEFAULT 0, "
+                        "PRIMARY KEY (queryId)) "
+                        "ENGINE = MEMORY";
+    LOGS(_log, LOG_LVL_DEBUG, "Executing query: " << query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGS(_log, LOG_LVL_ERROR, "SQL query failed: " << query);
+        throw SqlError(ERR_LOC, errObj);
+    }
+    trans.commit();
+}
+
+
+void QStatusMysql::queryStatsTmpRegister(QueryId queryId, int totalChunks) {
+    std::lock_guard<std::mutex> sync(_dbMutex);
+    QMetaTransaction trans(_conn);
+    sql::SqlErrorObject errObj;
+    // sql::SqlResults results; &&&
+    std::string query = "INSERT INTO QStatsTmp (queryId, totalChunks, completedChunks, queryBegin, lastUpdate) "
+                        "VALUES ( ";
+    query += std::to_string(queryId) + ", ";
+    query += std::to_string(totalChunks) + ", ";
```

Comment:
CC files could be made more readable by eliminating `std::` using `using namespace std;` at the beginning of the file.

Reply:
Then the query generation could be rewritten in one line:
```
auto const query =
    "INSERT INTO QStatsTmp (queryId, totalChunks, completedChunks, queryBegin, lastUpdate) "
    "VALUES ("+ to_string(queryId) + "," + to_string(totalChunks) + ", " + "0, NOW(), NOW())";
```